controlplane/src/core/repositories/FederatedGraphRepository.ts

```diff
@@ -246,27 +354,89 @@ export class FederatedGraphRepository {
     return result[0]?.count || 0;
   }
 
-  public async targetByName(name: string): Promise<Target | undefined> {
+  public async byTargetId(targetId: string): Promise<FederatedGraphDTO | undefined> {
     const resp = await this.db.query.targets.findFirst({
       where: and(
-        eq(schema.targets.name, name),
+        eq(schema.targets.id, targetId),
         eq(schema.targets.organizationId, this.organizationId),
         eq(schema.targets.type, 'federated'),
       ),
+      with: {
+        federatedGraph: {
+          with: {
+            subgraphs: {
+              columns: {
+                subgraphId: true,
+              },
+            },
+          },
+        },
+        namespace: true,
+        labelMatchers: true,
+      },
     });
 
     if (!resp) {
       return undefined;
     }
 
-    return resp;
+    const latestVersion = await this.db
```

Comment:
It feels like this should already exists and can be reused from a different repo or service.

Reply:
Yes, we use the same in `byName`

Reply:
Refactored